src/Version.h

```diff
@@ -17,26 +17,79 @@
 
 namespace Version {
 
-using exact_width_mumble_raw_version_t = std::uint32_t;
-using mumble_raw_version_t             = unsigned int;
+enum CompareMode { AtLeast, LessThan };
+
+//
+// The mumble version format (v2) is a uint64 and consists of the following:
+// reserved/unused  major   minor   patch
+// 0xFFFF           0xFFFF  0xFFFF  0xFFFF
```

Comment:
I am wondering whether it might be more future-proof to use the rightmost 2 bytes as the reserved/unused ones. That way we could eventually switch to a 4-component version number (where the new version component would be less important than the patch number), if we wanted to, without breaking backwards compatibility :thinking:

Reply:
@Hartmnt this is still something that I think should be adapted